spec/cohortmanager.spec.js

```diff
@@ -0,0 +1,316 @@
+
+    // verify
+    expect(result).toEqual(expected)
+
+    })
+
+it("Cohorts can't exist without a name", ()  => {
+    // set up
+    const cohortManager = new CohortManager()
+    const expected = false
+
+    // execute
+    const result = cohortManager.addCohort("", 2)
+
+    // verify
+    expect(result).toEqual(expected)
+
+    })
+
+it("same student can't exist in multiple cohorts", ()  => {
+    // set up
+    const cohortManager = new CohortManager()
+    const expected = false
+
+    // execute
+    cohortManager.addCohort("cohortOne", 2)
+    cohortManager.addCohort("cohortTwo", 2)
+    cohortManager.addStudentToCohort("cohortOne", "Ngolo", "Kante", "kantecodes", "kantecodes@gmail.com")
+    const result = cohortManager.addStudentToCohort("cohortTwo", "Ngolo", "Kante", "kantecodes", "kantecodes@gmail.com")
+
+    // verify
+    expect(result).toEqual(expected)
+    })
+
+it("remove student that exists from a cohort that they are not in", ()  => {
+    // set up
+    const cohortManager = new CohortManager()
+    const expected = false
+    // execute
+    cohortManager.addCohort("cohortOne", 2)
+    cohortManager.addStudentToCohort("cohortOne", "Ngolo", "Kante", "kantecodes", "kantecodes@gmail.com")
+    cohortManager.addStudentToCohort("cohortTwo", "Reece", "James", "reececodes", "reececodes@gmail.com")
+    const result = cohortManager.removeStudentFromCohort("cohortOne", "reececodes@gmail.com")
+    // verify
+    expect(result).toEqual(expected)
+    })
+
+it("Search for students by first name and surname and return all matching results", ()  => {
+    // set up
+    const cohortManager = new CohortManager()
+    const expected = [new Student("cohortOne", "Ngolo", "Kante", "kantecodes", "kantecodes@gmail.com", 1), new Student("cohortOne", "Dan", "Kante", "dankcodes", "dankcodes@gmail.com", 4), new Student("cohortOne", "Reece", "Kante", "rkcodes", "rkcodes@gmail.com", 5), new Student("cohortTwo", "Ngolo", "Smith", "nscodes", "nscodes@gmail.com", 8)]
+    // execute
+    cohortManager.addCohort("cohortOne", 5)
+    cohortManager.addCohort("cohortTwo", 5)
+    cohortManager.addStudentToCohort("cohortOne", "Ngolo", "Kante", "kantecodes", "kantecodes@gmail.com")
+    cohortManager.addStudentToCohort("cohortOne", "Reece", "James", "reececodes", "reececodes@gmail.com")
+    cohortManager.addStudentToCohort("cohortOne", "Dan", "James", "jamescodes", "jamescodes@gmail.com")
+    cohortManager.addStudentToCohort("cohortOne", "Dan", "Kante", "dankcodes", "dankcodes@gmail.com")
+    cohortManager.addStudentToCohort("cohortOne", "Reece", "Kante", "rkcodes", "rkcodes@gmail.com")
+    cohortManager.addStudentToCohort("cohortTwo", "Adam", "Smith", "adamscodes", "adamscodes@gmail.com")
+    cohortManager.addStudentToCohort("cohortTwo", "Adam", "Jones", "adamjcodes", "adamjcodes@gmail.com")
+    cohortManager.addStudentToCohort("cohortTwo", "Ngolo", "Smith", "nscodes", "nscodes@gmail.com")
```

Comment:
This looks a bit repetitive and long. I wonder if JS has a concise syntax to code this bit 🤔